lib/Doctrine/DBAL/Connection.php

```diff
@@ -1332,6 +1291,18 @@ public function executeCacheQuery($sql, $params, $types, QueryCacheProfile $qcp)
 
         $stmt->setFetchMode($this->defaultFetchMode);
 
+        return $this->createV3CompatStatement($stmt);
+    }
+
+    /**
+     * @return ResultStatement&V3Result
+     */
+    private function createV3CompatStatement(ResultStatement $stmt)
+    {
+        if (! $stmt instanceof V3Result) {
```

Comment:
Negation can be avoided to improve readability:
```php
if ($stmt instanceof V3Result) {
    return $stmt;
}

return new V3CompatStatement($stmt);
```

Reply:
Done